p2p/net/udp.go

```diff
@@ -81,14 +90,76 @@ func newUDPListener(listenAddress *net.UDPAddr) (*net.UDPConn, error) {
 
 var IPv4LoopbackAddress = net.IP{127, 0, 0, 1}
 
+func (n *UDPNet) initSession(to string, remote p2pcrypto.PublicKey) (NetworkSession, error) {
+	raddr, err := resolveUDPAddr(to)
+	if err != nil {
+		return nil, err
+	}
+
+	session := createSession(n.local.PrivateKey(), remote)
+	handshakeMessage, err := generateHandshakeMessage(session, n.config.NetworkID, 1010, n.local.PublicKey())
+	if err != nil {
+		return nil, err
+	}
+	_, err = n.conn.WriteToUDP(handshakeMessage, raddr)
```

Comment:
Do we need to wait for the handshake response? Can the local node start send data after it sent the handshake request, but before it received the response?

Reply:
If the answer is no, where do you store the fact that the session is waiting for an HS response

Reply:
Currently, the session we're using doesn't wait for a response. the initiator knows both the keys and creates a shared secret with it. when the receiver receives the message with the initiator public key, he can create a shared secret with it and his own key.